torch/csrc/utils/python_arg_parser.h

```diff
@@ -671,10 +671,20 @@ inline c10::optional<at::ScalarType> PythonArgs::scalartypeOptional(int i) {
   return scalartype(i);
 }
 
+inline at::Layout toLayout(PyObject* obj) {
+  if (THPLayout_Check(obj)) {
+    const auto layout = reinterpret_cast<THPLayout*>(obj);
+    return layout->layout;
+  }
+  const auto layout = THPUtils_unpackLong(obj);
+  TORCH_CHECK(layout >= 0, "Layout must not be negative");
+  return static_cast<at::Layout>(layout);
```

Comment:
This is the thing where `__torch_dispatch__` gives you an integer instead of the layout object, right? This should get fixed by https://github.com/pytorch/pytorch/pull/77543/files Do you still need this logic?

Reply:
Additionally, this doesn't do any checking that the integer corresponds to a real layout

Reply:
Yeah I think you're right - I don't think it's needed, was put there more just in case. I'll remove it!